src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -683,6 +683,23 @@ protected override void ProcessRecord()
         /// </summary>
         protected override void StopProcessing() => _cancelToken?.Cancel();
 
+        /// <summary>
+        /// Cleanup disposable WebSession if it is not being kept between invocations.
+        /// </summary> 
+        protected override void EndProcessing()
+        {
+            base.EndProcessing();
+            if (!IsPersistentSession())
+            {
+                WebSession?.Dispose();
```

Comment:
Cmdlet class has Dispose(). So we could move this in that overload.

Reply:
I can only see an internal InternalDispose() method which is not virtual. I'm unsure what I should do here to not cause a leak. I was fairly certain EndProcessing was marked as the cleanup method for CmdLets which is why I implemented here.

Could you provide more info on how I would implement the Dispose() - should I make the cmdlet implement IDisposable or override the InternalDispose() method?

Reply:
Sorry, I suggest to make the cmdlet IDisposable.

Reply:
ok, have done this.